source/acquaman/AMGenericStepScanConfiguration.cpp

```diff
@@ -283,16 +283,18 @@ void AMGenericStepScanConfiguration::setRegionOfInterestBoundingRange(AMRegionOf
 
 void AMGenericStepScanConfiguration::setI0(const AMDetectorInfo &info)
 {
+	bool validInfo = (info.name() != "Invalid detector");
```

Comment:
It should be in the class of `AMDetectorInfo`, since only `AMDetectorInfo` knows what is valid or invalid.